code/game/jobs/job_exp.dm

```diff
@@ -0,0 +1,197 @@
+		return 1
+	if(!job.exp_requirements || !job.exp_type)
+		return 1
+	if(!job_is_xp_locked(rank))
+		return 1
+	if(!M.client)
+		return 0
+	if(check_rights(R_ADMIN, 0, M))
+		return 1
+	var/list/play_records = params2list(M.client.prefs.exp)
+	var/isexempt = text2num(play_records["Exempt"])
+	if(isexempt)
+		return 1
+	var/my_exp = text2num(play_records[job.exp_type])
+	var/job_requirement = text2num(job.exp_requirements)
+	if(config.use_exp_restrictions_heads_hours && ((rank in command_positions) || rank == "AI"))
+		job_requirement = text2num(config.use_exp_restrictions_heads_hours) * 60
+	if(my_exp >= job_requirement)
+		return 1
+	else
+		return 0
+
+/proc/job_is_xp_locked(jobtitle)
+	if(!config.use_exp_restrictions_heads && ((jobtitle in command_positions) || jobtitle == "AI"))
+		return 0
+	if(!config.use_exp_restrictions_other && !((jobtitle in command_positions) || jobtitle == "AI"))
+		return 0
+	return 1
+
+
+/mob/proc/get_exp_report()
+	if(client)
+		return client.get_exp_report()
+	else
+		return "[src] has no client."
+
+/client/proc/get_exp_report()
+	if(!config.use_exp_tracking)
+		return "Tracking is disabled in the server configuration file."
+
+	var/list/play_records = params2list(prefs.exp)
+	if(!play_records.len)
+		return "[src] has no records."
+
+	var/return_text = ""
+
+	var/list/exp_data = list()
+	for(var/cat in exp_jobsmap)
+		if(text2num(play_records[cat]))
+			exp_data[cat] = text2num(play_records[cat])
+		else
+			exp_data[cat] = 0
+
+	for(var/dep in exp_data)
+		if(exp_data[dep] > 0)
+			if(dep == "Exempt")
+				return_text += "<BR>- Exempt (all jobs auto-unlocked)"
+			else
+				return_text += "<BR>- [dep] " + get_exp_format(exp_data[dep]) + " (" + num2text(round(exp_data[dep]/exp_data["Crew"]*100)) + "%)"
+
+	var/list/jobs_locked = list()
+	var/list/jobs_unlocked = list()
+	for(var/datum/job/job in job_master.occupations)
+		if(job.exp_requirements && job.exp_type)
+			if(!job_is_xp_locked(job.title))
+				continue
+			else if(has_exp_for_job(mob, job.title))
+				jobs_unlocked += "<BR>- " + job.title
+			else
+				var/xp_req = job.exp_requirements
+				if(config.use_exp_restrictions_heads_hours && ((job.title in command_positions) || job.title == "AI"))
+					xp_req = text2num(config.use_exp_restrictions_heads_hours) * 60
+				jobs_locked += "<BR>- " + job.title + " (" + get_exp_format(text2num(play_records[job.exp_type])) + " / " + get_exp_format(xp_req) + " [job.exp_type] EXP)"
+	if(jobs_unlocked.len)
+		return_text += "<BR><BR>Jobs Unlocked: "
+		for(var/text in jobs_unlocked)
+			return_text += text
+	if(jobs_locked.len)
+		return_text += "<BR><BR>Jobs Not Unlocked: "
+		for(var/text in jobs_locked)
+			return_text += text
+	return return_text
+
+
+/client/proc/get_exp_general()
+	var/list/play_records = params2list(prefs.exp)
+	var/exp_gen = text2num(play_records["Crew"])
+	return get_exp_format(exp_gen)
+
+/client/proc/get_exp_format(var/expnum)
+	if(expnum > 0)
+		return num2text(round(expnum / 60)) + "h"
```

Comment:
Thought about it, but decided against it, since the extra detail doesn't add anything. All numbers less than an hour equate to "very new" and never unlock anything.